toolkit/tools/internal/targetos/targetos.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package targetos
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/envfile"
+)
+
+type TargetOs string
+
+const (
+	TargetOsAzureLinux2 TargetOs = "azl2"
+	TargetOsAzureLinux3 TargetOs = "azl3"
+)
+
+func GetInstalledTargetOs(rootfs string) (TargetOs, error) {
+	fields, err := envfile.ParseEnvFile(filepath.Join(rootfs, "etc/os-release"))
+	if err != nil {
+		return "", fmt.Errorf("failed to read /etc/os-release file:\n%w", err)
+	}
+
+	distroId := fields["ID"]
+	versionId := fields["VERSION_ID"]
+
+	switch distroId {
+	case "mariner":
+		switch versionId {
+		case "2.0":
+			return TargetOsAzureLinux2, nil
+
+		default:
+			return "", fmt.Errorf("unknown VERSION_ID (%s) for CBL-Mariner in /etc/os-release", versionId)
+		}
+
+	case "azurelinux":
```

Comment:
hmm I have mixed feelings about this. Do we want to have a default behavior instead of erroring out of unsupported versions? (this can also happen at the caller, but here seems more central)
My concern is around maintainability. For a new OS version, someone will need to update this flow.

Reply:
I think erroring out is probably a good thing. When a new distro/version is released, we will want to add a new set of filesystem options for that distro/version.

I guess in theory we could have some sort of universal default. That could be more interesting, if and when we decide to support other distros. But I don't think it is needed for now.

Reply:
My thinking is that a dev working on azl 4.0 upbringing will be blocked on this.